File: TopInterview150/Math/PlusOne/solution.py
class Solution(object):
    def plusOne(self, digits):
        """
        :type digits: List[int]
        :rtype: List[int]
        """
        for i in range(len(digits)-1, -2, -1):
            if i == -1:
                ret = [0]*(len(digits)+1)
                ret[0] = 1
                return ret
            if digits[i] != 9:
                digits[i] += 1
                return digits
            digits[i] = 0

File: TopInterview150/Math/PlusOne/solution.cpp
class Solution {
public:
    vector<int> plusOne(vector<int>& digits) {
        for (int i = digits.size()-1; i > -2; i--)
        {
            if (i == -1)
            {
                vector<int> ret(digits.size() + 1, 0);
                ret[0] = 1;
                return ret;            
            }
            if (digits[i] != 9)
            {
                digits[i]++;
                return digits;
            }
            digits[i] = 0;
        }
        return digits;
    }
};